dot/rpc/modules/chain_test.go

```diff
@@ -2,3 +2,494 @@
+				mockBlockAPI,
+			},
+			args: args{
+				req: &ChainHashRequest{&testHash},
+			},
+			exp: expRes,
+		},
+		{
+			name: "GetHeader ERR",
+			fields: fields{
+				mockBlockAPIErr,
+			},
+			args: args{
+				req: &ChainHashRequest{&testHash},
+			},
+			expErr: errors.New("GetFinalisedHash Error"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cm := &ChainModule{
+				blockAPI: tt.fields.blockAPI,
+			}
+			res := ChainBlockHeaderResponse{}
+			err := cm.GetHeader(tt.args.r, tt.args.req, &res)
+			if tt.expErr != nil {
+				assert.EqualError(t, err, tt.expErr.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, tt.exp, res)
+		})
+	}
+}
+
+func TestChainModule_ErrSubscriptionTransport(t *testing.T) {
+	req := &EmptyRequest{}
+	res := &ChainBlockHeaderResponse{}
+	cm := NewChainModule(new(apimocks.BlockAPI))
+
+	err := cm.SubscribeFinalizedHeads(nil, req, res)
+	require.Equal(t, err, ErrSubscriptionTransport)
+
+	err = cm.SubscribeNewHead(nil, req, res)
+	require.Equal(t, err, ErrSubscriptionTransport)
+
+	err = cm.SubscribeNewHeads(nil, req, res)
+	require.Equal(t, err, ErrSubscriptionTransport)
+}
+
+func TestHeaderToJSON(t *testing.T) {
+	emptyHeader := types.NewEmptyHeader()
+	vdts := types.NewDigest()
+	err := vdts.Add(
+		types.PreRuntimeDigest{
+			ConsensusEngineID: types.BabeEngineID,
+			Data:              common.MustHexToBytes("0x0201000000ef55a50f00000000"),
+		},
+		types.ConsensusDigest{
+			ConsensusEngineID: types.BabeEngineID,
+			Data:              common.MustHexToBytes("0x0118ca239392960473fe1bc65f94ee27d890a49c1b200c006ff5dcc525330ecc16770100000000000000b46f01874ce7abbb5220e8fd89bede0adad14c73039d91e28e881823433e723f0100000000000000d684d9176d6eb69887540c9a89fa6097adea82fc4b0ff26d1062b488f352e179010000000000000068195a71bdde49117a616424bdc60a1733e96acb1da5aeab5d268cf2a572e94101000000000000001a0575ef4ae24bdfd31f4cb5bd61239ae67c12d4e64ae51ac756044aa6ad8200010000000000000018168f2aad0081a25728961ee00627cfe35e39833c805016632bf7c14da5800901000000000000000000000000000000000000000000000000000000000000000000000000000000"),
+		},
+		types.SealDigest{
+			ConsensusEngineID: types.BabeEngineID,
+			Data:              common.MustHexToBytes("0x4625284883e564bc1e4063f5ea2b49846cdddaa3761d04f543b698c1c3ee935c40d25b869247c36c6b8a8cbbd7bb2768f560ab7c276df3c62df357a7e3b1ec8d"),
+		},
+	)
+	require.NoError(t, err)
+
+	header, err := types.NewHeader(common.Hash{}, common.Hash{}, common.Hash{}, big.NewInt(21), vdts)
+	require.NoError(t, err)
+
+	expRes, err := HeaderToJSON(*header)
+	require.NoError(t, err)
+	expResEmpty, err := HeaderToJSON(*emptyHeader)
+	require.NoError(t, err)
+	type args struct {
+		header types.Header
+	}
+	tests := []struct {
+		name string
+		args args
+		exp  ChainBlockHeaderResponse
+	}{
+		{
+			name: "HeaderToJSON Empty Header",
+			args: args{
+				header: *emptyHeader,
+			},
+			exp: expResEmpty,
+		},
+		{
+			name: "HeaderToJSON NonEmpty Header",
```

Comment:
Just call it `not empty`. The parent test name already contains the `HeaderToJSON` string. Same for above test case as well.